pkgs/development/compilers/blueprint/default.nix

```diff
@@ -29,18 +30,32 @@ stdenv.mkDerivation rec {
 
   buildInputs = [
     python3
-  ];
+    gtk4
+  ] ++ (with python3.pkgs; [
+    pygobject3
+    wrapPython
+  ]);
 
   propagatedBuildInputs = [
-    # So that the compiler can find GIR and .ui files
     gobject-introspection
   ];
 
+  postFixup = ''
+    makeWrapperArgs="\
+      --prefix GI_TYPELIB_PATH : $GI_TYPELIB_PATH \
```

Comment:
Why add this and GTK 4? I would expect the typelibs to be picked up from the app’s build environment. Inheriting it from blueprint will make it harder to override GTK as well.

Reply:
I had the following error: 
```
Traceback (most recent call last):
  File "/home/bene/git/nixpkgs/./result/bin/blueprint-compiler", line 37, in <module>
    from blueprintcompiler import main
  File "/nix/store/45rfii5jp79c0k5yijxawwix6xpk2cd8-blueprint-compiler-0.6.0/lib/python3.10/site-packages/blueprintcompiler/main.py", line 25, in <module>
    from .lsp import LanguageServer
  File "/nix/store/45rfii5jp79c0k5yijxawwix6xpk2cd8-blueprint-compiler-0.6.0/lib/python3.10/site-packages/blueprintcompiler/lsp.py", line 24, in <module>
    from .completions import complete
  File "/nix/store/45rfii5jp79c0k5yijxawwix6xpk2cd8-blueprint-compiler-0.6.0/lib/python3.10/site-packages/blueprintcompiler/completions.py", line 22, in <module>
    from . import gir, language
  File "/nix/store/45rfii5jp79c0k5yijxawwix6xpk2cd8-blueprint-compiler-0.6.0/lib/python3.10/site-packages/blueprintcompiler/gir.py", line 24, in <module>
    import gi # type: ignore
ModuleNotFoundError: No module named 'gi'
```

I searched for the error and found [this](https://github.com/NixOS/nixpkgs/issues/25493)  with your comment. I just used and it worked. So it needed gtk4 to get the version test to work. Maybe you can explain your suggested implementation a bit more.

Reply:
Yes. Cool.